src/SamplesApp/UITests.Shared/Microsoft_UI_Xaml_Controls/RatingControlTests/RatingControlPage.xaml.cs

```diff
@@ -0,0 +1,635 @@
+using RatingItemImageInfo = Microsoft.UI.Xaml.Controls.RatingItemImageInfo;
+using RatingItemFontInfo = Microsoft.UI.Xaml.Controls.RatingItemFontInfo;
+using Uno.UI.Samples.Controls;
+
+namespace UITests.Microsoft_UI_Xaml_Controls.RatingControlTests
+{
+	[Sample("RatingControl", "WinUI")]
+	public sealed partial class RatingControlPage : Page
+	{
+		UIElement _secondTextBlockUI;
+		DispatcherTimer _dt;
+
+		SolidColorBrush _tomato = new SolidColorBrush(Colors.Tomato);
+		SolidColorBrush _aqua = new SolidColorBrush(Colors.Aqua);
+
+		public RatingControlPage()
+		{
+			this.InitializeComponent();
+
+			TestRatingControl.ValueChanged += TestRatingControl_ValueChanged;
+
+			RatingDarkTheme.PlaceholderValue = 1.5;
+
+			MyRatingReadOnlyTextBlock.Text = "2.2";
+			MyRatingReadOnlyWithPlaceholder.PlaceholderValue = 3.3;
+
+			DisabledWithValue.Value = 3;
+			DisabledWithPlaceholderValue.PlaceholderValue = 3;
+
+			CustomImages.Value = 3.0;
+			CustomImages.PlaceholderValue = 1.5;
+
+			var imageInfo = new RatingItemImageInfo();
+			imageInfo.Image = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_set.png"));
+			imageInfo.UnsetImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_unset.png")); ;
+			imageInfo.PlaceholderImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_placeholder.png")); ;
+			imageInfo.DisabledImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_disabled.png")); ;
+			imageInfo.PointerOverImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_mouseoverset.png")); ;
+			imageInfo.PointerOverPlaceholderImage = new Windows.UI.Xaml.Media.Imaging.BitmapImage(new Uri("ms-appx:/Assets/rating_mouseoverplaceholder.png"));
+
+			CustomImagesTwo.ItemInfo = imageInfo;
+			CustomImagesTwo.Value = 3.0;
+			CustomImagesTwo.PlaceholderValue = 4.25;
+
+			PointerOverPlaceholderFallbackRating.AddHandler(RatingControl.PointerMovedEvent, new PointerEventHandler(PointerOverPlaceholderFallbackRating_PointerMoved), true);
+			PointerOverFallbackRating.AddHandler(RatingControl.PointerMovedEvent, new PointerEventHandler(PointerOverFallbackRating_PointerMoved), true);
+
+			PointerOverPlaceholderImageFallbackRating.AddHandler(RatingControl.PointerMovedEvent, new PointerEventHandler(PointerOverPlaceholderImageFallbackRating_PointerMoved), true);
+			PointerOverImageFallbackRating.AddHandler(RatingControl.PointerMovedEvent, new PointerEventHandler(PointerOverImageFallbackRating_PointerMoved), true);
+
+			ColorFlipButton.Foreground = _tomato;
+
+			RatingBindingSample.DataContext = CaptionStringBox;
+			BindingRatingCaption.DataContext = ColorFlipButton;
+
+			//TODO: Uno Platform - WinUI wraps controls in a "TestFrame" which has some "ambient" features, including the view scaling checkbox
+			//var testFrame = Window.Current.Content as TestFrame;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6162540)